wazo_agid/handlers/tests/test_outgoing_callerid_formatter.py

```diff
@@ -158,3 +158,87 @@ def test_selected_valid_raw_name_preserved(self) -> None:
             'CALLERID(all)',
             '"Foobar" <123>',
         )
+
+    def test_diversion_going_national(self) -> None:
+        channel_vars = {
+            dv.SELECTED_CALLER_ID: '+14189990000',
+            dv.TRUNK_CID_FORMAT: 'national',
+            dv.DST_REDIRECTING_EXTERN_NUM: '+14189990000',
+            'WAZO_TENANT_COUNTRY': 'CA',
+            'WAZO_DST_REDIRECTING_EXTERN_NAME': 'Test',
+        }
+        self.agi.get_variable.side_effect = channel_vars.get
+
+        self.handler.execute()
+
+        print(self.agi.set_variable.mock_calls)
+
+        self.agi.set_variable.assert_has_calls(
+            [
+                call('REDIRECTING(from-num,i)', '4189990000'),
+                call('REDIRECTING(from-name,i)', 'Test'),
+            ]
+        )
+
+    def test_diversion_going_e164(self) -> None:
+        channel_vars = {
+            dv.SELECTED_CALLER_ID: '+14189990000',
+            dv.TRUNK_CID_FORMAT: 'E164',
+            dv.DST_REDIRECTING_EXTERN_NUM: '+14189990000',
+            'WAZO_TENANT_COUNTRY': 'CA',
+            'WAZO_DST_REDIRECTING_EXTERN_NAME': 'Test',
+        }
+        self.agi.get_variable.side_effect = channel_vars.get
+
+        self.handler.execute()
+
+        print(self.agi.set_variable.mock_calls)
+
+        self.agi.set_variable.assert_has_calls(
+            [
+                call('REDIRECTING(from-num,i)', '14189990000'),
+                call('REDIRECTING(from-name,i)', 'Test'),
+            ]
+        )
+
+    def test_diversion_going_plus_E164(self) -> None:
+        channel_vars = {
+            dv.SELECTED_CALLER_ID: '+14189990000',
+            dv.TRUNK_CID_FORMAT: '+E164',
+            dv.DST_REDIRECTING_EXTERN_NUM: '+14189990000',
+            'WAZO_TENANT_COUNTRY': 'CA',
+            'WAZO_DST_REDIRECTING_EXTERN_NAME': 'Test',
+        }
+        self.agi.get_variable.side_effect = channel_vars.get
+
+        self.handler.execute()
+
+        print(self.agi.set_variable.mock_calls)
+
+        self.agi.set_variable.assert_has_calls(
+            [
+                call('REDIRECTING(from-num,i)', '+14189990000'),
+                call('REDIRECTING(from-name,i)', 'Test'),
+            ]
+        )
+
+    def test_diversion_going_extension(self) -> None:
+        channel_vars = {
+            dv.SELECTED_CALLER_ID: '+14189990000',
+            dv.TRUNK_CID_FORMAT: '+E164',
+            dv.DST_REDIRECTING_EXTERN_NUM: '1000',
+            'WAZO_TENANT_COUNTRY': 'CA',
+            'WAZO_DST_REDIRECTING_EXTERN_NAME': 'Test',
+        }
+        self.agi.get_variable.side_effect = channel_vars.get
+
+        self.handler.execute()
+
+        print(self.agi.set_variable.mock_calls)
+
+        self.agi.set_variable.assert_has_calls(
+            [
+                call('REDIRECTING(from-num,i)', '1000'),
+                call('REDIRECTING(from-name,i)', 'Test'),
+            ]
+        )
```

Comment:
I think integration tests should also be updated.
https://github.com/wazo-platform/wazo-agid/blob/0ebd3de07b1b93e02c8bd6d4ec7ba430de8ac3ac/integration_tests/suite/test_handlers.py#L699-L711